src/components/DatePicker/index.tsx

```diff
@@ -0,0 +1,32 @@
+import { DatePicker, DatePickerProps } from '@mui/x-date-pickers/DatePicker';
+import React from 'react';
+import { Controller } from 'react-hook-form';
+
+type Props = DatePickerProps<string> & {
+  minDate: string;
+  maxDate: string;
+};
+const CustomDatePicker: React.FC<Props> = ({
+  minDate,
+  maxDate,
+  ...datePickerProps
+}) => {
+  return (
+    <Controller
+      name="selectedDate"
+      render={({ field }) => (
+        <DatePicker<string>
+          label="Select Date"
+          format="MM/dd/yyyy"
+          value={field.value}
+          onChange={(date) => field.onChange(date)}
+          minDate={minDate}
+          maxDate={maxDate}
+          {...datePickerProps}
+        />
+      )}
+    />
+  );
+};
```

Comment:
`onChange` jest taki domyślny więc jak nic więcej nie robi to jest tutaj zbędny

```suggestion
    <Controller
      name={name}
      render={({ field }) => (
        <DatePicker<string>
          label={label}
          format="YYYY-MM-DD"
          value={field.value}
          minDate={minDate}
          maxDate={maxDate}
          {...datePickerProps}
        />
      )}
    />
  );
};
```